src/App.tsx

```diff
@@ -5,29 +5,61 @@ import AddItemForm from './components/AddItemForm';
 import TaskDashboard from './components/TaskDashboard';
 import TaskItem from './components/TaskItem';
 import { TaskItemType } from './types/TaskItemType';
+import EditTaskModal from './components/EditTaskModal';
 
 function App() {
   const [taskList, setTaskList] = useState<TaskItemType[]>([]);
 
-  const handleAddTask = (taskTitle: string) => {
+  const [showEditModal, setShowEditModal] = useState<boolean>(false);
+
+  const [taskTitle, setTaskTitle] = useState<string>('');
+
+  const [taskId, setTaskId] = useState<number>(0);
+
+  const handleAddTask = (inputText: string) => {
     setTaskList([
       ...taskList,
       {
         id: taskList.length ? taskList[taskList.length - 1].id + 1 : 0,
-        title: taskTitle,
+        title: inputText,
         done: false,
       },
     ]);
   };
 
+  const handleEditTask = (
+    selectedTaskTitle: string,
+    selectedTaskId: number,
+  ) => {
+    setShowEditModal(true);
+    setTaskTitle(selectedTaskTitle);
+    setTaskId(selectedTaskId);
+  };
+
+  const handleCloseModal = () => {
+    setShowEditModal(false);
+  };
+
+  const handleSaveTask = (editedTaskTitle: string) => {
+    taskList.filter((item) => item.id === taskId)[0].title = editedTaskTitle;
+  };
+
   return (
     <div className="todoapp stack-large">
       <Title />
       <AddItemForm onAddTask={handleAddTask} />
       <TaskDashboard />
+
       {taskList.map((item) => (
-        <TaskItem taskData={item} />
+        <TaskItem taskData={item} onEdit={handleEditTask} />
```

Comment:
The app is still showing the error:
![image](https://user-images.githubusercontent.com/14007745/206908379-1947e03a-fbe1-4c31-b87f-6845fcc18b07.png)

I believe is due to this line, if you have any doubts please let me know!